ui/selectors/signatures.ts

```diff
@@ -0,0 +1,30 @@
+import { createSelector } from 'reselect';
+import { DefaultRootState } from 'react-redux';
+import {
+  unapprovedMsgsSelector,
+  unapprovedPersonalMsgsSelector,
+  unapprovedTypedMessagesSelector,
+} from './transactions';
+import { createDeepEqualSelector } from './util';
+
+export const selectUnapprovedMessages = createSelector(
+  unapprovedMsgsSelector,
+  unapprovedPersonalMsgsSelector,
+  unapprovedTypedMessagesSelector,
+  (messages, personalMsgs, typedMessages) => ({
+    ...messages,
+    ...personalMsgs,
+    ...typedMessages,
+  }),
+);
+
+const internalSelectUnapprovedMessage = createSelector(
+  selectUnapprovedMessages,
+  (_state: DefaultRootState, messageId: string) => messageId,
+  (messages, messageId) => messages[messageId],
+);
+
+export const selectUnapprovedMessage = createDeepEqualSelector(
+  internalSelectUnapprovedMessage,
+  (message) => message,
+);
```

Comment:
These new selectors aren't really specific to confirmations, since they are simply querying the signature controller state and using `createSelector` and `createDeepEqualSelector` to minimise re-renders. All the existing message selectors were legacy and mixed with transactions.

On a broader level, I worry that creating separate confirmation specific selectors might restrict their re-usability since all the existing selectors are defined at the top level in `ui/selectors`, so arguably any truly confirmation specific ones could also sit in `ui/selectors/confirmation.ts`?